test/e2e/ratelimit/ratelimit_universal.go

```diff
@@ -67,43 +73,62 @@ var _ = E2EBeforeSuite(func() {
 })
 
 func RateLimitOnUniversal() {
-	verifyRateLimit := func(client string, total int) func() int {
-		return func() int {
+	// In the best case, we start sending requests and can send all successfully
+	// and consecutively before the bucket is empty.
+	// Otherwise the bucket empties before we've sent all which will lead to the
+	// number of successes being more than the limit.
+	// So we send requests until the rate limit is first hit, send failing
+	// requests until the first one succeeds, then we send as
+	// many requests as we can until we are rate limited again.
+	// We compare that number of successes to the expected number.
+	// The assumption here is that we can send enough requests to hit the limit
+	// within the interval.
+	verifyRateLimit := func(client string, svc string, expected int) {
+		verifyError := fmt.Sprintf("couldn't verify rate limit for %s to %s to equal %d", client, svc, expected)
```

Comment:
verifyErrorMsg would easier to understand also it might help readability if this is defined in the `verify` function